src/PayID/pay-id-client.ts

```diff
@@ -22,7 +24,45 @@ export default class PayIDClient {
    * @param payID The payID to resolve for an address.
    * @returns An XRP address representing the given PayID if one exists, otherwise undefined.
    */
-  public async xrpAddressForPayID(_payID: string): Promise<string | undefined> {
-    throw PayIDError.unimplemented
+  public async xrpAddressForPayID(payID: string): Promise<string | undefined> {
+    const paymentPointer = PayIDUtils.parsePaymentPointer(payID)
+    if (!paymentPointer) {
+      throw PayIDError.invalidPaymentPointer
+    }
+
+    const url = `https://${paymentPointer.host}${paymentPointer.path}`
+    // TODO(keefertaylor): Attach network parameter.
+    // TODO(keefertaylor): Generalize the below to allow requesting in other types.
+    const requestConfig = {
+      headers: {
+        Accept: 'application/xrp+json',
+      },
+    }
+
+    return this.axiosInstance
```

Comment:
Why promise structure here versus async await?

Reply:
I actually would rather use `await` but it threw exceptions which sucked because then I had to type them. Do you have an example of this in action?

Reply:
I think we can type them as AxiosError. I can pull this down and check though

Reply:
Try catch method of async await
```ts
    try {
      const response = await this.axiosInstance.get<{ address: string }>(
        url,
        requestConfig,
      )
      if (!response) return
      const { address } = response.data
      if (!address) {
        throw new PayIDError(
          PayIDErrorType.UnexpectedResponse,
          'Sucessful response was in an unknown format',
        )
      }
      return address
    } catch (error) {
      // Handle erroneous responses from the server.
      if (error.response) {
        // 404 means no mapping was found.
        if (error.response.status === 404) {
          return undefined
        }

        // Otherwise re-throw the error as an unexepected response.
        throw new PayIDError(
          PayIDErrorType.UnexpectedResponse,
          `${error.response.status} ${error.response.statusText}: ${error.response.data}`,
        )
      } else {
        // Generically errors from the server which contained no response (timeouts, etc).
        throw new PayIDError(PayIDErrorType.Unknown, error.message)
      }
    }
```

or

catch model (saves a level of indentation and reads easier IMO)
```ts
    const response = await this.axiosInstance
      .get<{ address: string }>(url, requestConfig)
      .catch((error) => {
        // Handle erroneous responses from the server.
        if (error.response) {
          // 404 means no mapping was found.
          if (error.response.status === 404) {
            return undefined
          }

          // Otherwise re-throw the error as an unexepected response.
          throw new PayIDError(
            PayIDErrorType.UnexpectedResponse,
            `${error.response.status} ${error.response.statusText}: ${error.response.data}`,
          )
        } else {
          // Generically errors from the server which contained no response (timeouts, etc).
          throw new PayIDError(PayIDErrorType.Unknown, error.message)
        }
      })
    if (!response) return
    const { address } = response.data
    if (!address) {
      throw new PayIDError(
        PayIDErrorType.UnexpectedResponse,
        'Sucessful response was in an unknown format',
      )
    }
    return address
```

Reply:
I agree both of these are better. In my original implementation (and both of yours) the throw from the unexpected response format gets caught in the catch, which misbehaves. 

The only way around this I can find is to wrap the whole thing in a promise, which I've done but which is really ugly.